lib/rules/disallow-function-prototype-extension.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+var includes = require("../utils/includes");
+var extensionFixes = {
+  'property': "Use the computed('dependentKey', function() {}) syntax instead.",
+  'observes': "Use the observer('dependentKey', function() {}) syntax instead.",
+  'observesBefore': "Also, using before observers is deprecated."
```

Comment:
`Also, using before observers is deprecated.` <- IMO this is not relevant to the linter. For example, someone may choose to ignore that deprecation in the late 1.x series, and it would be weird to be telling them it is deprecated again though they ignored it.

Basically I think this should scope to linting, and not jump into deprecations for which we already have a fairly mature solution.